homeworks/12/chat-system/src/Client.java

```diff
@@ -0,0 +1,99 @@
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Client {
+    private Socket socket;
+    private BufferedReader bufferedReader;
+    private BufferedWriter bufferedWriter;
+    private String nickname;
+
+    public Client(Socket socket, String nickname) {
+        try {
+            this.socket = socket;
+            bufferedReader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            bufferedWriter = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream()));
+            this.nickname = nickname;
+        } catch (IOException exception){
+            try {
+                bufferedReader.close();
+                bufferedWriter.close();
+                socket.close();
+            }catch (Exception exception1) {
+                System.out.println(exception1);
+            }
+        }
+    }
+
+    public void sendMessage() {
+        try {
+            bufferedWriter.write(nickname);
+            bufferedWriter.newLine();
+            bufferedWriter.flush();
+
+            Scanner scanner = new Scanner(System.in);
+            while (socket.isConnected()) {
+                String message = scanner.nextLine();
+                if(message.startsWith("/msg")) {
+                    String newString = message.substring("/msg".length());
+                    bufferedWriter.write(nickname + ": " + newString);
+                    bufferedWriter.newLine();
+                    bufferedWriter.flush();
+                }
+            }
+        } catch (IOException exception) {
+            try {
+                bufferedReader.close();
+                bufferedWriter.close();
+                socket.close();
+            }catch (Exception exception1) {
+                System.out.println(exception1);
+            }
+        }
+    }
+
+    public void listenForMessage() {
+        new Thread(() -> {
+            String msgFromGroupChat;
+
+            while (socket.isConnected()) {
+                try {
+                    msgFromGroupChat = bufferedReader.readLine();
+                    System.out.println(msgFromGroupChat);
+                } catch (IOException exception) {
+                    try {
+                        bufferedReader.close();
+                        bufferedWriter.close();
+                        socket.close();
+                    }catch (Exception exception1) {
+                        System.out.println(exception1);
+                    }
+                }
+            }
+        }).start();
+    }
+
+    public static void main(String[] args) throws IOException {
+        Scanner scanner = new Scanner(System.in);
+        String nickname = null;
+        String host = null;
+        int port = 0;
+        Socket socket = new Socket();
+        while(host == null || port == 0 || nickname == null) {
+            String command = scanner.nextLine();
+            if (command.startsWith("/nick")) {

```

Comment:
You do not let the user to change their nickname after setting it for the first time when joining the server, which makes the `/nick`  command pretty useless in my opinion.